app/scripts/metamask-controller.js

```diff
@@ -308,6 +309,13 @@ module.exports = class MetamaskController extends EventEmitter {
     }
   }
 
+  setLocked () {
+    this.keyringController.setLocked()
+      .then((data) => {
+        this.platform.reload()
```

Comment:
That fits with our current design mantra that "the metamask controller is not platform dependent".